packages/extension-vscode/src/quickFixProvider.ts

```diff
@@ -0,0 +1,77 @@
+            return `Ignore hint '${category}' in this project`;
+        }],
+        ['vscode-webhint/ignore-hint-global', (category: string) => {
+            return `Ignore hint '${category}' in all projects`;
+        }],
+        ['vscode-webhint/ignore-problem-project', (name: string) => {
+            return `Ignore problem '${name}' in this project`;
+        }],
+        ['vscode-webhint/ignore-problem-global', (name: string) => {
+            return `Ignore problem '${name}' in all projects`;
+        }]
+    ]);
+
+    public constructor(documents: TextDocuments<TextDocument>, sourceName: string) {
+        this.documents = documents;
+        this.sourceName = sourceName;
+    }
+
+    public provideCodeActions(params: CodeActionParams): CodeAction[] | null {
+        const textDocument = this.documents.get(params.textDocument.uri);
+
+        if (!textDocument) {
+            return null;
+        }
+
+        const results: CodeAction[] = [];
+
+        params.context.diagnostics.forEach((currentDiagnostic) => {
+            this.quickFixActionsTemplates.forEach((value, key) => {
+
+                // only respond to Microsoft Edge Tools requests.
+                if (!currentDiagnostic.source || currentDiagnostic.source !== this.sourceName) {
+                    return;
+                }
+
+                // default title
+                let title = currentDiagnostic.code?.toString() || '';
+
+                // if it is a problem (not a hint) use the problem as a title instead
+                if (key.includes('-problem-')) {
+                    const problemName = getProblemNameFromDiagnostic(currentDiagnostic);
+
+                    title = problemName ? problemName : title;
+                }
+
+                // create custom codeActions and associating custom commands.
+                const customCommand: Command = { command: key, title };
+
+                customCommand.arguments = [title, currentDiagnostic.code];
+                const customCodeAction = CodeAction.create(value(title), customCommand, CodeActionKind.QuickFix);
+
+                customCodeAction.diagnostics = [currentDiagnostic];
+                results.push(customCodeAction);
+            });
+        });
+
+        const editCurrentProjectConfigTitle = 'Edit .hintrc for current project';
+        const editGlobalProjectConfigTitle = 'Edit .hintrc for all projects';
+        const editCurrentProjectConfig: Command = { command: 'vscode-webhint/edit-hintrc-project', title: editCurrentProjectConfigTitle };
+        const editGlobalProjectConfig: Command = { command: 'vscode-webhint/edit-hintrc-global', title: editGlobalProjectConfigTitle };
+
+        results.push(CodeAction.create(editCurrentProjectConfigTitle, editCurrentProjectConfig, CodeActionKind.QuickFix));
+        results.push(CodeAction.create(editGlobalProjectConfigTitle, editGlobalProjectConfig, CodeActionKind.QuickFix));
```

Comment:
The addition of these entries needs to be conditional based on whether we've actually reported a problem at a given location. Currently they cause the lightbulb to show up no matter where in the document I've put the cursor, even if there's no issue at that location:
![image](https://user-images.githubusercontent.com/785009/168397210-94de9982-ef18-4b10-a40a-f0485b1862cb.png)